src/modules/tas_optimizer/mod.rs

```diff
@@ -331,10 +333,19 @@ fn optim_init(marker: MainThreadMarker, path: PathBuf, first_frame: usize) {
 
     // TODO: get current parameters.
     let parameters = unsafe {
+        let game_dir = CStr::from_ptr(engine::com_gamedir.get(marker).cast())
+            .to_str()
+            .unwrap();
+
+        let is_paranoia: bool = game_dir.eq("paranoia");
+        let is_cstrike: bool = game_dir.eq("cstrike");
+        let is_czero: bool = game_dir.eq("czero");
```

Comment:
We don't want to `com_gamedir.get()` here because it'll panic if `com_gamedir` wasn't found. Instead, in that case we want to assume that there are no game-dir-related effects.
```suggestion
        // Safety: the reference does not outlive this block, and com_gamedir can only be modified
        // at engine start and while setting the HD models or the addon folder.
        let game_dir = engine::com_gamedir
            .get_opt(marker)
            .and_then(|dir| CStr::from_ptr(dir.cast()).to_str().ok())
            .unwrap_or("valve");

        let is_paranoia = game_dir == "paranoia";
        let is_cstrike = game_dir == "cstrike";
        let is_czero = game_dir == "czero";
```